File: frontend2/src/models/Coupon.ts
import Category from "./Category";
import Company from "./Company";

type CouponConstractorArguments = {
  id?: number;
  title: string;
  description: string;
  amount: number;
  price: number;
  startDate: Date;
  endDate: Date;
  company?: Company;
  category?: Category;
};

class Coupon {
  public readonly id?: number;
  public title: string;
  public description: string;
  public amount: number;
  public price: number;
  public startDate: Date;
  public endDate: Date;
  public company?: Company;
  public category?: Category;

  constructor(args: CouponConstractorArguments) {
    this.title = args.title;
    this.description = args.description;
    this.startDate = args.startDate;
    this.endDate = args.endDate;
    this.amount = args.amount;
    this.price = args.price;
    this.company = args.company;
    this.category = args.category;
  }
}
export default Coupon;

File: src/main/java/com/nya/mitzi/coupon/Coupon.java
package com.nya.mitzi.coupon;

import com.nya.mitzi.company.Company;
import com.nya.mitzi.couponCustomer.CouponCustomer;
import jakarta.persistence.*;
import com.nya.mitzi.category.Category;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@Table(name="coupons")
@Entity
public class Coupon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    @Column(name="title")
    private String title;
    @Column(name="description")
    private String description;
    @Column(name="start_date")
    private LocalDate startDate;
    @Column(name="end_date")
    private LocalDate endDate;
    @Column(name="amount")
    private Integer amount;
    @Column(name="price")
    private double price;
    @Column(name="image")
    private String image;

    @ManyToOne()
    @JoinColumn(name="company_id")
    @ToString.Exclude
    private Company company;

    @ManyToOne(cascade = CascadeType.REMOVE, fetch = FetchType.EAGER)
    @JoinColumn(name="category_id")
    private Category category;

    @ToString.Exclude
    @OneToMany(mappedBy = "coupon", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    private List<CouponCustomer> couponCustomer;



}
